### PR TITLE
subversion: add non-homebrew universal python fail

### LINKED ISSUE #34119: Subversion fails on 10.10 --with-python
Brewing subversion fails when including --with-python. The failure occurs during `make swig-py` because one of the link stages is trying to create a universal binary even though all previous stages (plus dependencies) were only built for x86_64.

Note (at the bottom of this post) the two lines that begin with `cd subversion...`. The first one invokes libtool with `--mode=link /usr/bin/clang ... -march=native`, but the second one uses `--mode=link cc ... -arch x86_64 -arch i386` and fails because the dependencies do not have 32-bit versions.

https://gist.github.com/ca19cd2ee70a2a9b27e1

If you look in the gist [here](https://gist.github.com/anonymous/ca19cd2ee70a2a9b27e1#file-config-log-L1460) you can see that `SWIG_PY_COMPILE` was altered by the formula:

```ruby
    # The system Python is built with llvm-gcc, so we override this
    # variable to prevent failures due to incompatible CFLAGS
    ENV['ac_cv_python_compile'] = ENV.cc
```

However, `ac_cv_python_link` is not overriden, so it still contains its reference to cc. The original definitions come from subversion-1.8.10/build/get-py-info.py, which normally would set

```
SWIG_PY_COMPILE = "cc -arch x86_64 -arch i386 -pipe"
SWIG_PY_LINK = "cc -bundle -undefined dynamic_lookup -arch x86_64 -arch i386 -Wl,-F. -framework Python"
```

Partial log output follows:

```
==> make swig-py
/bin/sh /private/tmp/subversion-v0hxQz/subversion-1.8.10/libtool --tag=CC --silent --mode=compile /usr/bin/clang -Werror=unknown-warning-option -Os -w -pipe -march=native -mmacosx-version-min=10.10 -O2 -Wall  -DNDEBUG  -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK -DDARWIN_10  -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/usr/local/Cellar/apr/1.5.1/include/apr-1  -I/usr/include/apr-1  -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/python/libsvn_swig_py -prefer-pic -c -o subversion/bindings/swig/python/svn_client.lo subversion/bindings/swig/python/svn_client.c
/bin/sh /private/tmp/subversion-v0hxQz/subversion-1.8.10/libtool --tag=CC --silent --mode=compile /usr/bin/clang -Werror=unknown-warning-option -Os -w -pipe -march=native -mmacosx-version-min=10.10 -O2 -Wall  -DNDEBUG -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK -DDARWIN_10   -DSWIGPYTHON -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/usr/local/Cellar/apr/1.5.1/include/apr-1  -I/usr/include/apr-1  -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/python/libsvn_swig_py -I./subversion/include -I./subversion -I/usr/local/Cellar/apr/1.5.1/include/apr-1   -I/usr/include/apr-1 -I/usr/local/Cellar/subversion/1.8.10_1/libexec/serf/include/serf-1 -I/usr/local/opt/sqlite/include -I/usr/include -o subversion/bindings/swig/python/libsvn_swig_py/swigutil_py.lo -c subversion/bindings/swig/python/libsvn_swig_py/swigutil_py.c
/bin/sh /private/tmp/subversion-v0hxQz/subversion-1.8.10/libtool --tag=CC --silent --mode=compile /usr/bin/clang -Werror=unknown-warning-option -Os -w -pipe -march=native -mmacosx-version-min=10.10 -O2 -Wall  -DNDEBUG  -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK -DDARWIN_10  -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/usr/local/Cellar/apr/1.5.1/include/apr-1  -I/usr/include/apr-1  -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/python/libsvn_swig_py -prefer-pic -c -o subversion/bindings/swig/python/core.lo subversion/bindings/swig/python/core.c
/bin/sh /private/tmp/subversion-v0hxQz/subversion-1.8.10/libtool --tag=CC --silent --mode=compile /usr/bin/clang -Werror=unknown-warning-option -Os -w -pipe -march=native -mmacosx-version-min=10.10 -O2 -Wall  -DNDEBUG  -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK -DDARWIN_10  -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/usr/local/Cellar/apr/1.5.1/include/apr-1  -I/usr/include/apr-1  -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/python/libsvn_swig_py -prefer-pic -c -o subversion/bindings/swig/python/svn_delta.lo subversion/bindings/swig/python/svn_delta.c
/bin/sh /private/tmp/subversion-v0hxQz/subversion-1.8.10/libtool --tag=CC --silent --mode=compile /usr/bin/clang -Werror=unknown-warning-option -Os -w -pipe -march=native -mmacosx-version-min=10.10 -O2 -Wall  -DNDEBUG  -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK -DDARWIN_10  -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/usr/local/Cellar/apr/1.5.1/include/apr-1  -I/usr/include/apr-1  -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/python/libsvn_swig_py -prefer-pic -c -o subversion/bindings/swig/python/svn_diff.lo subversion/bindings/swig/python/svn_diff.c
/bin/sh /private/tmp/subversion-v0hxQz/subversion-1.8.10/libtool --tag=CC --silent --mode=compile /usr/bin/clang -Werror=unknown-warning-option -Os -w -pipe -march=native -mmacosx-version-min=10.10 -O2 -Wall  -DNDEBUG  -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK -DDARWIN_10  -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/usr/local/Cellar/apr/1.5.1/include/apr-1  -I/usr/include/apr-1  -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/python/libsvn_swig_py -prefer-pic -c -o subversion/bindings/swig/python/svn_fs.lo subversion/bindings/swig/python/svn_fs.c
/bin/sh /private/tmp/subversion-v0hxQz/subversion-1.8.10/libtool --tag=CC --silent --mode=compile /usr/bin/clang -Werror=unknown-warning-option -Os -w -pipe -march=native -mmacosx-version-min=10.10 -O2 -Wall  -DNDEBUG  -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK -DDARWIN_10  -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/usr/local/Cellar/apr/1.5.1/include/apr-1  -I/usr/include/apr-1  -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/python/libsvn_swig_py -prefer-pic -c -o subversion/bindings/swig/python/svn_ra.lo subversion/bindings/swig/python/svn_ra.c
/bin/sh /private/tmp/subversion-v0hxQz/subversion-1.8.10/libtool --tag=CC --silent --mode=compile /usr/bin/clang -Werror=unknown-warning-option -Os -w -pipe -march=native -mmacosx-version-min=10.10 -O2 -Wall  -DNDEBUG  -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK -DDARWIN_10  -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/usr/local/Cellar/apr/1.5.1/include/apr-1  -I/usr/include/apr-1  -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/python/libsvn_swig_py -prefer-pic -c -o subversion/bindings/swig/python/svn_repos.lo subversion/bindings/swig/python/svn_repos.c
/bin/sh /private/tmp/subversion-v0hxQz/subversion-1.8.10/libtool --tag=CC --silent --mode=compile /usr/bin/clang -Werror=unknown-warning-option -Os -w -pipe -march=native -mmacosx-version-min=10.10 -O2 -Wall  -DNDEBUG  -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK -DDARWIN_10  -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/include -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/proxy -I/usr/local/Cellar/apr/1.5.1/include/apr-1  -I/usr/include/apr-1  -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/python/libsvn_swig_py -prefer-pic -c -o subversion/bindings/swig/python/svn_wc.lo subversion/bindings/swig/python/svn_wc.c
mkdir /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/bindings/swig/python/libsvn
cd subversion/bindings/swig/python/libsvn_swig_py && /bin/sh /private/tmp/subversion-v0hxQz/subversion-1.8.10/libtool --tag=CC --silent --mode=link /usr/bin/clang  -Werror=unknown-warning-option -Os -w -pipe -march=native -mmacosx-version-min=10.10 -O2 -Wall  -DNDEBUG  -L/usr/local/opt/openssl/lib -L/usr/local/opt/sqlite/lib -L/usr/local/opt/readline/lib -L/usr/local/lib -Wl,-headerpad_max_install_names   -L/usr/local/Cellar/subversion/1.8.10_1/libexec/serf/lib   -rpath /usr/local/Cellar/subversion/1.8.10_1/lib  -o libsvn_swig_py-1.la  swigutil_py.lo ../../../../../subversion/libsvn_client/libsvn_client-1.la ../../../../../subversion/libsvn_wc/libsvn_wc-1.la ../../../../../subversion/libsvn_ra/libsvn_ra-1.la ../../../../../subversion/libsvn_delta/libsvn_delta-1.la ../../../../../subversion/libsvn_subr/libsvn_subr-1.la -laprutil-1 -L/usr/local/Cellar/apr/1.5.1/lib -lapr-1 -framework CoreFoundation -framework Security -framework CoreServices
cd subversion/bindings/swig/python && /bin/sh /private/tmp/subversion-v0hxQz/subversion-1.8.10/libtool --tag=CC --silent --mode=link cc -bundle -undefined dynamic_lookup -arch x86_64 -arch i386 -Wl,-F. -framework Python -L/usr/local/opt/openssl/lib -L/usr/local/opt/sqlite/lib -L/usr/local/opt/readline/lib -L/usr/local/lib -Wl,-headerpad_max_install_names  -rpath /usr/local/Cellar/subversion/1.8.10_1/lib/svn-python/libsvn -avoid-version -module  -o _core.la  core.lo ../../../../subversion/bindings/swig/python/libsvn_swig_py/libsvn_swig_py-1.la ../../../../subversion/libsvn_diff/libsvn_diff-1.la ../../../../subversion/libsvn_subr/libsvn_subr-1.la -L/usr/local/Cellar/apr/1.5.1/lib -lapr-1 -framework CoreFoundation -framework Security -framework CoreServices
ld: warning: ignoring file .libs/core.o, file was built for unsupported file format ( 0xCF 0xFA 0xED 0xFE 0x07 0x00 0x00 0x01 0x03 0x00 0x00 0x00 0x01 0x00 0x00 0x00 ) which is not the architecture being linked (i386): .libs/core.o
ld: warning: ignoring file ../../../../subversion/bindings/swig/python/libsvn_swig_py/.libs/libsvn_swig_py-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): ../../../../subversion/bindings/swig/python/libsvn_swig_py/.libs/libsvn_swig_py-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_client/.libs/libsvn_client-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_client/.libs/libsvn_client-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_wc/.libs/libsvn_wc-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_wc/.libs/libsvn_wc-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_diff/.libs/libsvn_diff-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_diff/.libs/libsvn_diff-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_ra/.libs/libsvn_ra-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_ra/.libs/libsvn_ra-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_ra_local/.libs/libsvn_ra_local-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_ra_local/.libs/libsvn_ra_local-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_repos/.libs/libsvn_repos-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_repos/.libs/libsvn_repos-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_fs/.libs/libsvn_fs-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_fs/.libs/libsvn_fs-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_fs_fs/.libs/libsvn_fs_fs-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_fs_fs/.libs/libsvn_fs_fs-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_fs_util/.libs/libsvn_fs_util-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_fs_util/.libs/libsvn_fs_util-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_ra_svn/.libs/libsvn_ra_svn-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_ra_svn/.libs/libsvn_ra_svn-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_ra_serf/.libs/libsvn_ra_serf-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_ra_serf/.libs/libsvn_ra_serf-1.dylib
ld: warning: ignoring file /usr/local/Cellar/subversion/1.8.10_1/libexec/serf/lib/libserf-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /usr/local/Cellar/subversion/1.8.10_1/libexec/serf/lib/libserf-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_delta/.libs/libsvn_delta-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_delta/.libs/libsvn_delta-1.dylib
ld: warning: ignoring file ../../../../subversion/libsvn_diff/.libs/libsvn_diff-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): ../../../../subversion/libsvn_diff/.libs/libsvn_diff-1.dylib
ld: warning: ignoring file /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_subr/.libs/libsvn_subr-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /private/tmp/subversion-v0hxQz/subversion-1.8.10/subversion/libsvn_subr/.libs/libsvn_subr-1.dylib
ld: warning: ignoring file ../../../../subversion/libsvn_subr/.libs/libsvn_subr-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): ../../../../subversion/libsvn_subr/.libs/libsvn_subr-1.dylib
ld: warning: ignoring file /usr/local/Cellar/apr/1.5.1/lib/libapr-1.dylib, file was built for x86_64 which is not the architecture being linked (i386): /usr/local/Cellar/apr/1.5.1/lib/libapr-1.dylib
ld: in '/usr/local/Cellar/apr/1.5.1/lib/libapr-1.0.dylib', file was built for x86_64 which is not the architecture being linked (i386): /usr/local/Cellar/apr/1.5.1/lib/libapr-1.0.dylib for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [subversion/bindings/swig/python/_core.la] Error 1
```

### PR DESCRIPTION
**Pull request by DomT4**
Borrows my change from https://github.com/Homebrew/homebrew-versions/pull/779

Closes #34119

### CODE REVIEW

File: Library/Formula/subversion.rb [hunk 1]
```diff
@@ -95,7 +107,7 @@ def install
     end
 
     if build.include? "unicode-path"
-      fail <<-EOS.undent
+      raise <<-EOS.undent
```
Comment (DomT4) [inline Library/Formula/subversion.rb]: `brew audit --strict` said I should change this.

**Comment by DomT4** [comment]
CC @mikemcquaid @tdsmith for :+1: / :-1:

**Comment by mikemcquaid** [comment]
:+1:

File: Library/Formula/subversion.rb [hunk 2]
```diff
@@ -70,6 +70,18 @@ class Subversion < Formula
   end
 
   def install
+    # OS X's Python is built universally and can't link with Homebrew's deps
+    # unless Homebrew's deps are universal as well.
+    # https://github.com/Homebrew/homebrew-versions/issues/777
+    if build.with?("python") && !File.exist?(HOMEBREW_PREFIX/"bin/python")
+      unless build.universal?
```
Comment (tdsmith) [inline Library/Formula/subversion.rb]: I think the condition you want to test is

`if build.with?("python") && (which "python").universal? && !build.universal?`

File: Library/Formula/subversion.rb [hunk 3]
```diff
@@ -70,6 +70,18 @@ class Subversion < Formula
...
+        raise <<-EOS.undent
+          You must build subversion17 --universal unless Homebrew's
```
Comment (tdsmith) [inline Library/Formula/subversion.rb]: just "subversion"

Comment (DomT4) [inline Library/Formula/subversion.rb]: Oops. Someone can kick me for forgetting to change this line when I copied it across :smile_cat:

**Comment by tdsmith** [comment]
It would also be fine to patch subversion's build script to strip `-arch i386` from everything unless universal is set. If this formula used superenv, superenv would take care of it automatically.

The python stuff has an explicit `-framework Python`; it would be nice to strip that as well.

**Comment by tdsmith** [comment]
(nice, but not blocking)

File: Library/Formula/subversion.rb (hunk 2 above)
Comment (DomT4) [inline Library/Formula/subversion.rb]: I was mimicking [this block](https://github.com/DomT4/homebrew/blob/svn/Library/Formula/subversion.rb#L121-L133) further down the formula, but I'm happy to change if you're cool with it?

Comment (tdsmith) [inline Library/Formula/subversion.rb]: I don't care if you write it as one `if` or an `if` and an `unless` but asking whether python is brewed is the wrong question; what you really care about is whether python is universal or not.

Comment (DomT4) [inline Library/Formula/subversion.rb]: I presumed OS X's Python would always be universal, and we weren't keen on supporting the Python we warn about in `brew doctor` so I worded it like this to try and "enforce" a "friendly" Python. But I'm cool with the change - I'd pretty much jump through a glass window before going against one of your Python-handling recommendations :smile_cat:

**Comment by DomT4** [comment]
> It would also be fine to patch subversion's build script to strip -arch i386 from everything unless universal is set.

I wasn't sure if this would be too invasive for Homebrew, but I guess it wouldn't be doing anything `superenv` doesn't in a different way?

>  it would be nice to strip that as well.

[Something similar to this](https://gist.githubusercontent.com/tdsmith/9026da299ac1bfd3f419/raw/b73a919c38af08941487ca37d46e711864104c4d/boost-python.diff) or is it likely to be significantly different? Happy to try.

File: Library/Formula/subversion.rb (hunk 2 above)
Comment (tdsmith) [inline Library/Formula/subversion.rb]: Aw, thanks! (I'm not sure anyone actually does, but you _can_ `brew install python --universal`.)

**Comment by tdsmith** [comment]
Yeah, the build scripts are probably invoking `python-config --ldflags` and it should just use `-undefined dynamic_lookup` instead.

And it would be nice to symlink the contents of /usr/local/Cellar/subversion/1.8.13/lib/svn-python into lib/python2.7/site-packages... :)

**Comment by DomT4** [comment]
`subversion` enjoys burying all the little "find-xyz" components in various scripts, which makes this kinda fun, but this seems to be the place where they're pulling in the Framework:

```python
  # Initialize config variables
  assert os.name == "posix"
  options = sysconfig.get_config_var('LDSHARED').split()
  fwdir = sysconfig.get_config_var('PYTHONFRAMEWORKDIR')

  if fwdir and fwdir != "no-framework":

    # Setup the framework prefix
    fwprefix = sysconfig.get_config_var('PYTHONFRAMEWORKPREFIX')
    if fwprefix != "/System/Library/Frameworks":
      add_option_if_missing(options, "-F%s" % fwprefix)

    # Load in the framework
    fw = sysconfig.get_config_var('PYTHONFRAMEWORK')
    add_option(options, "-framework", fw)

  elif sys.platform == 'darwin':

    # Load bundles from python
    python_exe = os.path.join(sysconfig.get_config_var("BINDIR"),
      sysconfig.get_config_var('PYTHON'))
    add_option_if_missing(options, "-bundle_loader", python_exe)
```

**Comment by DomT4** [comment]
Seem to have removed the Framework linkage, albeit with the world's most blunt-axe patch.

```
checking for linking Python extensions... clang -bundle -undefined dynamic_lookup -bundle_loader /usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/bin/python
checking for linking Python libraries... -bundle -undefined dynamic_lookup -bundle_loader /usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/bin/python
```

No stern warning after install about Framework linkage either.

Also linked the site-packages stuff:

```
~> ls -l /usr/local/lib/python2.7/site-packages | grep svn
lrwxr-xr-x   1 Dominyk wheel     68 Jun  6 03:05 libsvn -> ../../../Cellar/subversion/1.8.13/lib/python2.7/site-packages/libsvn
lrwxr-xr-x   1 Dominyk wheel     65 Jun  6 03:05 svn -> ../../../Cellar/subversion/1.8.13/lib/python2.7/site-packages/svn
```

**Comment by tdsmith** [comment]
Great! I think that's worth kicking upstream; their code for frameworks should never be helpful.

**Comment by DomT4** [comment]
Is it worth kicking the actual patch upstream or just a request to not link into a Python Framework on OS X?

Are you happy for this to :ship:?

**Comment by tdsmith** [comment]
I think yes to both; I can open the issue and argue the case if you want.

:shipit:

**Comment by DomT4** [comment]
I'd appreciate that - There's only so much Python stuff I can go into detail about before I go "I'll have to ask Tim about this later so I can reply properly" :smile_cat:.

**Pull request closed without merging**